### DISCUSSION TITLE
redis.pconnect.connection_limit is being ignored.

### DISCUSSION THREAD

**Discussion by cdeguzman** [phpredis/phpredis#2314] [Q&A] [answered]
Hi,

Why does redis.pconnect.connection_limit setting, looks like its being ignored by phpredis. I have set this setting to 30, redis.pconnect.pooling_enabled=1 and redis.clusters.persistent=1, and saw the  current connections to redis well over its set limit. The limit was set to 30 but the current connections in redis was way beyond that like 300+ 

<img width="1237" alt="Screen Shot 2023-01-07 at 11 27 02 AM" src="https://user-images.githubusercontent.com/4004015/211129462-87b2ac89-56b3-43a0-9ac1-99cf661b5ce7.png">

I am using RedisCluster class to connect:

```
$redis = new \RedisCluster(null, $seeds, $timeout, $readTimeout, true)
```

Am I missing something to make it work or how does it actually limit the connections?

Below is the is environment I am running:

Server php version (AWS EC2):
```
PHP Version => 7.0.32
Thread Safety => disabled

*php-fpm pool set to 30
```

Redis (AWS Elasticache):
```
Engine version: 6.2.6
Cluster mode: On
```

Phpredis settings:
```
redis

Redis Support => enabled
Redis Version => 4.3.0
Available serializers => php, igbinary
Available compression => lzf

Directive => Local Value => Master Value
redis.arrays.algorithm => no value => no value
redis.arrays.auth => no value => no value
redis.arrays.autorehash => 0 => 0
redis.arrays.connecttimeout => 0 => 0
redis.arrays.consistent => 0 => 0
redis.arrays.distributor => no value => no value
redis.arrays.functions => no value => no value
redis.arrays.hosts => no value => no value
redis.arrays.index => 0 => 0
redis.arrays.lazyconnect => 0 => 0
redis.arrays.names => no value => no value
redis.arrays.pconnect => 0 => 0
redis.arrays.previous => no value => no value
redis.arrays.readtimeout => 0 => 0
redis.arrays.retryinterval => 0 => 0
redis.clusters.auth => no value => no value
redis.clusters.persistent => 1 => 1
redis.clusters.read_timeout => 0 => 0
redis.clusters.seeds => no value => no value
redis.clusters.timeout => 0 => 0
redis.pconnect.connection_limit => 30 => 30
redis.pconnect.pooling_enabled => 1 => 1
redis.session.lock_expire => 0 => 0
redis.session.lock_retries => 10 => 10
redis.session.lock_wait_time => 2000 => 2000
redis.session.locking_enabled => 0 => 0
```

**Comment by yatsukhnenko** [comment, marked as answer]
Hi @cdeguzman,
`redis.pconnect.connection_limit` is a per-worker limit of connections to the same server, so you may see up to 30 * N connections since you have configured 30 workers and N is the number of nodes in cluster.
